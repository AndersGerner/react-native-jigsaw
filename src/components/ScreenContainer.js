import * as React from "react";
import { SafeAreaView, StyleSheet, ScrollView, View } from "react-native";
import { withTheme } from "../core/theming";

class ScreenContainer extends React.Component {
  static defaultProps = {
    useThemeGutterSpacing: true
  };

  renderScrollableSafeAreaView(themeStyles) {
    const { children, style } = this.props;
    return (
      <SafeAreaView
        style={[
          styles.container,
          { backgroundColor: themeStyles.backgroundColor }
        ]}
      >
        <ScrollView
          style={themeStyles}
          contentContainerStyle={[
            {
              flexGrow: 1
            },
            style
          ]}
        >
          {children}
        </ScrollView>
      </SafeAreaView>
    );
  }

  renderSafeAreaView(themeStyles) {
    const { children, style } = this.props;
    return (
      <SafeAreaView
        style={[
          styles.container,
          { backgroundColor: themeStyles.backgroundColor }
        ]}
      >
        <View style={[styles.container, themeStyles, style]}>{children}</View>
      </SafeAreaView>
    );
  }

  renderScrollView(themeStyles) {
    const { children, style } = this.props;
    return (
      <ScrollView
        style={[themeStyles]}
        contentContainerStyle={[
          {
            flexGrow: 1
          },
          style
        ]}
      >
        {children}
      </ScrollView>
    );
  }

  renderView(themeStyles) {
    const { children, style } = this.props;
    return (
      <View style={[styles.container, themeStyles, style]}>{children}</View>
    );
  }

  render() {
    const {
      theme,
      hasSafeArea,
      scrollable,
      useThemeGutterSpacing
    } = this.props;

    const themeStyles = {
      backgroundColor: theme.colors.background
    };

    if (useThemeGutterSpacing) {
      themeStyles.paddingHorizontal = theme.spacing.gutters;
    }

    if (scrollable && hasSafeArea) {
      return this.renderScrollableSafeAreaView(themeStyles);
    }

    if (hasSafeArea) {
      return this.renderSafeAreaView(themeStyles);
    }

    if (scrollable) {
      return this.renderScrollView(themeStyles);
    }

    return this.renderView(themeStyles);
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
});

export default withTheme(ScreenContainer);
