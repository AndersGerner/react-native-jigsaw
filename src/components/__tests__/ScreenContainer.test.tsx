import * as React from "react";
import renderer from "react-test-renderer";
import ScreenContainer from "../ScreenContainer";

it("renders screen container", () => {
  const tree = renderer.create(<ScreenContainer />).toJSON();

  expect(tree).toMatchSnapshot();
});
