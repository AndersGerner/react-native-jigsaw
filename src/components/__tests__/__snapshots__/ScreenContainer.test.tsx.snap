// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders screen container 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "rgba(251, 252, 253, 1)",
      },
      undefined,
    ]
  }
/>
`;
